tests/plugins/test_sendmail.py

```diff
@@ -243,33 +244,42 @@ class WF(object):
             'from_address': 'foo@bar.com',
             'to_koji_submitter': to_koji_submitter,
             'to_koji_pkgowner': False,
-            'koji_hub': '',
+            'koji_hub': '/' if koji_integration else None,
             'koji_root': 'https://koji/',
             'koji_proxyuser': None,
             'koji_ssl_certs_dir': '/certs',
             'koji_krb_principal': None,
             'koji_krb_keytab': None
         }
         p = SendMailPlugin(None, WF(), **kwargs)
-        assert p.koji_root == 'https://koji'
-        subject, body, fail_logs = p._render_mail(autorebuild, False, auto_cancel, manual_cancel)
-
-        # full logs are only generated on a failed autorebuild
-        if not autorebuild or auto_cancel or manual_cancel:
-            assert not fail_logs
-        else:
-            assert fail_logs
 
         # Submitter is updated in _get_receivers_list
         try:
             p._get_receivers_list()
-        except Exception:
-            pass
+        except RuntimeError as ex:
+            # Only valid exception is a RuntimeError when there are no
+            # recipients available
+            assert str(ex) == 'No recipients found'
 
-        if to_koji_submitter:
-            subject, body, _ = p._render_mail(autorebuild, False, auto_cancel, manual_cancel)
+        subject, body, logs = p._render_mail(autorebuild, success,
+                                             auto_cancel, manual_cancel)
 
-        status = 'Canceled' if auto_cancel or manual_cancel else 'Failed'
+        if success:
+            # full logs are never generated on success
+            assert not logs
+        else:
+            # full logs are only generated on a failed autorebuild
+            if not autorebuild or auto_cancel or manual_cancel:
+                assert not logs
+            else:
+                assert logs
+
+        if auto_cancel or manual_cancel:
+            status = 'Canceled'
+        elif success:
+            status = 'Succeeded'
+        else:
+            status = 'Failed'
```

Comment:
Merge these two if statements, ie:
       if auto_cancel or manual_cancel:
            status = 'Canceled'
            assert not logs
        elif success:
            status = 'Succeeded'
            assert not logs
        else:
            status = 'Failed'
            if autorebuild:
                assert logs
            else:
                assert not logs